### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
⚠️ **Warning: this code was not tested with real hardware**

Changes:

- Fix an infinite loop on the state machine.
- Add a more expressive status display system with **only three** indicators.
- Change the pin mapping.

New pin mapping:

- Ultrasound sensor **trigger** pin => 2
- Ultrasound sensor **echo** pin => 3
- **Green** indicator => 4
- **Orange** indicator => 5
- **Red** indicator => 6

### CODE REVIEW

File: firmware/temperature_detection/temperature_detection.ino [hunk 1]
```diff
@@ -1,76 +1,73 @@
...
-const int ultrasound_trigger_pin = 9;
-const int ultrasound_echo_pin = 10;
-const int orange_indicator_pin = 11;
-const int red_indicator_pin = 12;
-const int green_indicator_pin = 13;
+const int ultrasound_trigger_pin = 2;
+const int ultrasound_echo_pin = 3;
+const int orange_indicator_pin = 4;
+const int red_indicator_pin = 5;
+const int green_indicator_pin = 6;
```
Comment (0x2b3bfa0) [inline firmware/temperature_detection/temperature_detection.ino]:
```suggestion
const int green_indicator_pin = 4;
const int orange_indicator_pin = 5;
const int red_indicator_pin = 6;
```

The indicator pin mapping was wrong

**Review by 1technophile — COMMENTED** [review]
Thanks for the PR, I was working on externalizing all the constant to a header file like a `user_config.h` , maybe we could take the opportunity of your PR to do it: 
```
#ifndef user_config_h
#define user_config_h

#define serial_monitor_speed    9600 // serial monitor speed
#define time_between_readings   250 // time between sensors readings in ms
#define time_before_leaving     250 // time between sensors readings in ms

// Pins definition
#define fever_indicator_pin     8
#define ultrasound_trigger_pin  9
#define ultrasound_echo_pin     10
#define orange_indicator_pin    11
#define red_indicator_pin       12
#define green_indicator_pin     13

#ifdef ESP32 // pin definition for ESP32 only
#define sda_pin     0
#define scl_pin     26
#endif

// Distances limits in cm
#define off_distance            30
#define dectection_distance     20
#define limit_distance          15

// Temperature limits
#define limit_fever             20

#endif
```

**Comment by 0x2b3bfa0** [comment]
Great 😄 Feel free to push to the [firmware-code-cleanup](https://github.com/helpfulengineering/project-temperature-detection/tree/firmware-code-cleanup) branch.

**Comment by 1technophile** [comment]
I'm going to test it on arduino Uno and M5stickc.

**Comment by 1technophile** [comment]
Test ok on Arduino UNO, I have added logs to make debugging easier. For me we can go forward and squash it.

Separated from this PR I think we should differentiate the fever LED from the distance one. It will be clearer in term of user understanding.

File: firmware/temperature_detection/temperature_detection.ino [hunk 2]
```diff
@@ -1,76 +1,75 @@
...
-    pinMode(fever_indicator_pin, OUTPUT);
     pinMode(green_indicator_pin, OUTPUT);
     pinMode(orange_indicator_pin, OUTPUT);
     pinMode(red_indicator_pin, OUTPUT);
+
+    Serial.begin(serial_monitor_speed);
+    
     #ifdef ESP32
-    Wire.begin(0,26);
+        Wire.begin(sda_pin,scl_pin);
```
Comment (0x2b3bfa0) [inline firmware/temperature_detection/temperature_detection.ino]:
```suggestion
        Wire.begin(sda_pin, scl_pin);
```
Add a space between parameters

**Comment by 0x2b3bfa0** [comment]
:shipit:

**Comment by 0x2b3bfa0** [comment]
> Separated from this PR I think we should differentiate the fever LED from the distance one. It will be clearer in term of user understanding.

I've removed it because having two indicators with the same color for different purposes might be a bad idea from the user experience viewpoint, however that is not my field of expertise and probably I was wrong. Feel free to revert that change.

**Pull request merged**